src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -509,7 +484,21 @@ bool DirectoryService::RunConsensusOnDSBlockWhenDSPrimary()
             + 1
         << "] BGIN");
 
-    cl->StartConsensus(PoWConsensusMessage, DSBlockHeader::SIZE);
+    auto announcementGeneratorFunc =
+        [this](vector<unsigned char>& dst, unsigned int offset,
+               const uint32_t consensusID,
+               const vector<unsigned char>& blockHash, const uint16_t leaderID,
+               const pair<PrivKey, PubKey>& leaderKey,
+               vector<unsigned char>& messageToCosign) mutable -> bool {
+        const auto& winnerPeer = m_allPoWConns.find(
```

Comment:
`m_allPoWConns` needs a mutex to be thread-safe

Reply:
The mutex `m_mutexAllPoWConns` at line 411 should still be held by the time this lambda is invoked inside `StartConsensus`